app/__init__.py

```diff
@@ -44,19 +44,61 @@ def products():
     if ("seller_email" not in session):
         return redirect('/login')
     products = Product.query.all()
-    print(products)
     return render_template('seller/products.html', products=products)
 
 @app.route('/products', methods=['POST'])
 def add_product():
     if ("seller_email" not in session):
         return redirect('/login')
-    print(request.form)
     new_product = Product(
         name=request.form['product_name'],
         price=float(request.form['product_price']),
         image_url=request.form['product_image'],
         category=request.form["product_selling_status"])
     db.session.add(new_product)
     db.session.commit()
-    return redirect('/products') 
\ No newline at end of file
+    return redirect('/products') 
+
+@app.route('/products/<id>', methods=['GET'])
+def get_product(id):
+    if ("seller_email" not in session):
+        return redirect('/login')
+    product = Product.query.get_or_404(id)
+    return render_template('seller/product.html', prod = product)
+
+@app.route('/products/<id>', methods=['POST'])
+def update_product(id):
+    if ("seller_email" not in session):
+        return redirect('/login')
+    product = Product.query.get_or_404(id)
+    product.name = request.form['product_name']
+    product.price = request.form['product_price']
+    product.image_url = request.form['product_image']
+    product.category = request.form['product_selling_status']
+    db.session.add(product)
+    db.session.commit()
+    return redirect(f'/products/{id}')
+
+@app.route('/products/<id>/subproducts', methods=['POST'])
+def add_subproduct(id):
+    if ("seller_email" not in session):
+        return redirect('/login')
+    new_subproduct = Subproduct(
+        product_id = id,
+        name = request.form['subproduct_name'],
+        image_url = request.form['subproduct_image']
+    )
+    db.session.add(new_subproduct)
+    db.session.commit()
+    return redirect(f'/products/{id}')
+
+@app.route('/products/<id>/subproducts/<sid>', methods=['POST'])
+def update_subproducts(id, sid):
+    if ("seller_email" not in session):
+        return redirect('/login')
+    subproduct = Subproduct.query.get(sid)
+    subproduct.name = request.form['subproduct_name']
+    subproduct.image_url = request.form['subproduct_image']
+    db.session.add(subproduct)
+    db.session.commit()
+    return redirect(f'/products/{id}')
```

Comment:
@ethanschreur This PR is meant to allow users to update products. I think this should be in a PR of its owm

Reply:
In my mind, part of editing a product is editing additional information about the product (like what subproducts are part of the product).

Reply:
> @ethanschreur This PR is meant to allow users to update products. I think this should be in a PR of its owm

In my mind, part of editing a product is editing additional information about the product (like what subproducts are part of the product).

Reply:
@ethanschreur being able to add a subproduct to a product its a feature on its own. Editing/Updating an already added subproduct is also a feature on its own.

Reply:
> @ethanschreur being able to add a subproduct to a product its a feature on its own. Being able to Edit/Update a already added subproduct is also a feature on its own.

I will make the changes